src/database.js

```diff
@@ -14,7 +14,7 @@ export class Database {
 
   all(collectionName) {
     this.checkFactoryPresence(collectionName);
-    return this.store[collectionName];
+    return _.cloneDeep(this.store[collectionName]);
```

Comment:
So, we only need to do it here since all the other accessor functions use `all`, right? 👍